infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/multiuser/oauth/OpenshiftTokenInitializationFilter.java

```diff
@@ -78,44 +79,54 @@ public OpenshiftTokenInitializationFilter(
   }
 
   @Override
-  protected String getUserId(String token) {
-    try {
-      io.fabric8.openshift.api.model.User user = getCurrentUser(token);
-      return firstNonNull(user.getMetadata().getUid(), user.getMetadata().getName());
-    } catch (KubernetesClientException e) {
-      if (e.getCode() == 401) {
-        LOG.error(
-            "Unauthorized when getting current user. Invalid OpenShift token, probably expired. Re-login? Re-request the token?");
-      }
-      throw new RuntimeException(e);
-    } catch (InfrastructureException e) {
-      throw new RuntimeException(e);
-    }
+  protected io.fabric8.openshift.api.model.User processToken(String token) {
+    return getCurrentUser(token);
+  }
+
+  @Override
+  protected String getUserId(String token, io.fabric8.openshift.api.model.User user) {
+    return firstNonNull(user.getMetadata().getUid(), user.getMetadata().getName());
   }
 
   @Override
-  protected Subject extractSubject(String token) {
+  protected @Nullable Subject extractSubject(
```

Comment:
is it really `@Nullable` ? Parent method is not and this looks like it can't return null either.